azurelinuxagent/common/logcollector.py

```diff
@@ -0,0 +1,360 @@
+]
+
+FILE_SIZE_LIMIT = 30 * 1024 * 1024  # 30 MB
+ARCHIVE_SIZE_LIMIT = 150 * 1024 * 1024  # 150 MB
+
+
+class LogCollector(object):
+
+    def __init__(self, manifest_file_path):
+        self.manifest_file_path = manifest_file_path
+        self.must_collect_files = self._expand_must_collect_files()
+
+    def _expand_must_collect_files(self):
+        # Match the regexes from the MUST_COLLECT_FILES list to existing file paths on disk.
+        manifest = []
+        for path in MUST_COLLECT_FILES:
+            expanded_paths = self._expand_path(path)
+            manifest.extend(expanded_paths)
+
+        return manifest
+
+    def _read_manifest_file(self):
+        return read_file(self.manifest_file_path).splitlines()
+
+    @staticmethod
+    def _log_to_results_file(entry):
+        if type(entry) == ustr or type(entry) == str:
+            append_file(OUTPUT_RESULTS_FILE_PATH, entry + "\n")
+        else:
+            append_items_to_file(OUTPUT_RESULTS_FILE_PATH, entry)
+
+    @staticmethod
+    def _expand_path(path):
+        # If the path needs expanding, e.g. '/var/log/azure/*', add all file matches, sorted by name.
+        # Otherwise, add the already expanded path, e.g. '/var/lib/waagent/HostingEnvironmentConfig.xml'
+        expanded = []
+        if os.path.exists(path):
+            expanded.append(path)
+        else:
+            paths = glob.glob(path)
+            if len(paths) > 0:
+                expanded.extend(sorted(paths))
+        return expanded
+
+    @staticmethod
+    def _parse_ll_command(folder):
+        output = run_command(["ls", "-alF", folder])
+        header = "Output of \"ll {0}\":\n".format(folder)
+        LogCollector._log_to_results_file(header + output)
+
+    @staticmethod
+    def _parse_echo_command(message):
+        LogCollector._log_to_results_file(message)
+
+    @staticmethod
+    def _parse_copy_command(path):
+        file_paths = LogCollector._expand_path(path)
+        LogCollector._log_to_results_file(file_paths)
+        return file_paths
+
+    @staticmethod
+    def _convert_file_name_to_archive(file_name):
+        if file_name.startswith(TRUNCATED_FILES_DIR):
+            # /var/truncated/var/log/syslog.1 becomes truncated_var_log_syslog.1
+            original_file_path = file_name[len(TRUNCATED_FILES_DIR):].lstrip(os.path.sep)
+            archive_file_name = "truncated_" + original_file_path.replace(os.path.sep, "_")
+            return archive_file_name
+        else:
+            return file_name.lstrip(os.path.sep)
+
+    @staticmethod
+    def _convert_archive_name_to_file_name(archive_name):
+        truncated_prefix = "truncated_"
+        if archive_name.startswith(truncated_prefix):
+            # truncated_var_log_syslog.1 becomes /var/truncated/var/log/syslog.1
+            file_name = archive_name[len(truncated_prefix):].replace("_", os.path.sep)
+            original_file_name = os.path.join(TRUNCATED_FILES_DIR, file_name.lstrip(os.path.sep))
```

Comment:
os.path.split instead?

Reply:
I'm joining paths here, not splitting. A file in the archive called `truncated_var_log_syslog.1` is actually `/var/truncated/var/log/syslog.1` on disk.

Reply:
Removed.